### PR TITLE
SSO Integration and Issue Fix#10

### PR DESCRIPTION
**Pull request by PramodVurilo**

### CODE REVIEW

**Comment by ChanKamWo** [comment]
@PramodVurilo please check 
image

Also, did you push all the changes? Are we expected to see some login options in UI
image

**Comment by PramodVurilo** [comment]
> @PramodVurilo please check image
> 
> Also, did you push all the changes? Are we expected to see some login options in UI image

Forntend implemention is not done yet.

File: AcademyKit.Server/Application/Common/Models/ResponseModels/OAuthTokenResponse.cs
Comment (ChanKamWo) [inline AcademyKit.Server/Application/Common/Models/ResponseModels/OAuthTokenResponse.cs:1]: Please follow C# naming convension in property names. use PascalCase

File: AcademyKit.Server/Infrastructure/Persistence/Migrations/20240816074649_isDefault col added in Group.cs
Comment (ChanKamWo) [inline AcademyKit.Server/Infrastructure/Persistence/Migrations/20240816074649_isDefault col added in Group.cs:1]: Class name should be in PascalCase

File: AcademyKit.Server/Api/Controllers/AccountController.cs [hunk 1]
```diff
@@ -232,5 +255,267 @@ await _changePasswordValidator
...
+
+        /// <summary>
+        /// sign in with microsoft
+        /// </summary>
+        /// <returns>the microsoft login interface</returns>
+        [HttpGet("signin-with-microsoft")]
+        [AllowAnonymous]
+        public IActionResult SignInWithMicrosoft()
+        {
+            var nonce = "Q2k4UWFtMmZ1NjlBNG1oRU1ENnNNRGhx";
```
Comment (ChanKamWo) [inline AcademyKit.Server/Api/Controllers/AccountController.cs:392]: Should nonce be generated on each call instead of hardcode?

**Review by ChanKamWo — CHANGES_REQUESTED** [review]
Some issues must be addressed before merging.

File: AcademyKit.Server/Api/Controllers/AccountController.cs [hunk 2]
```diff
@@ -232,5 +257,264 @@ await _changePasswordValidator
...
+        [AllowAnonymous]
+        public IActionResult SignInWithGoogle()
+        {
+            var url =
+                $"{_google.AuthUrl}"
+                + $"response_type=code&"
+                + $"client_id={_google.ClientId}&"
+                + $"scope=openid email&"
+                + $"redirect_uri={_google.RedirectUri}&"
+                + $"nonce={CommonHelper.GenerateNonce}&"
```
Comment (ChanKamWo) [inline AcademyKit.Server/Api/Controllers/AccountController.cs:275]: not a function call.

File: AcademyKit.Server/Api/Controllers/AccountController.cs [hunk 3]
```diff
@@ -232,5 +257,264 @@ await _changePasswordValidator
...
+        public IActionResult SignInWithMicrosoft()
+        {
+            var url =
+                $"{_microsoft.AuthUrl}"
+                + $"client_id={_microsoft.ClientId}&"
+                + $"response_type=code id_token&"
+                + $"redirect_uri={UrlEncoder.Default.Encode(_microsoft.RedirectUri)}&"
+                + $"response_mode=form_post&"
+                + $"scope=openid email profile User.Read offline_access&"
+                + $"nonce={CommonHelper.GenerateNonce}";
```
Comment (ChanKamWo) [inline AcademyKit.Server/Api/Controllers/AccountController.cs:401]: not a function call

File: AcademyKit.Server/Application/Common/Models/ResponseModels/OAuthTokenResponse.cs
Comment (ChanKamWo) [inline AcademyKit.Server/Application/Common/Models/ResponseModels/OAuthTokenResponse.cs:1]: Please follow PascalCase property naming convention. You can use `JsonProperty` attribute.

File: AcademyKit.Server/Infrastructure/Persistence/Migrations/20240816074649_isDefault col added in Group.cs
Comment (ChanKamWo) [inline AcademyKit.Server/Infrastructure/Persistence/Migrations/20240816074649_isDefault col added in Group.cs:1]: File name should follow PascalCase naming convention. with no whitespaces.

**Review by ChanKamWo — CHANGES_REQUESTED** [review]
Please merge main branch and fix the required issues.

**Pull request closed without merging**